client/blocks/site/index.jsx

```diff
@@ -117,11 +117,17 @@ class Site extends React.Component {
 			'is-compact': this.props.compact,
 		} );
 
-		// To ensure two Coming Soon badges don't appear while we introduce public coming soon
-		const isPublicComingSoon =
-			isEnabled( 'coming-soon-v2' ) && ! site.is_private && this.props.site.is_coming_soon;
+		// We show public coming soon badge only when the site is not private and the editing toolkit is available.
+		// Check for `! site.is_private` to ensure two Coming Soon badges don't appear while we introduce public coming soon.
+		const shouldPublicShowComingSoonSiteBadge =
```

Comment:
Tiny nit: should this be named `shouldShowPublicComingSoonSiteBadge` (re-ordering public and show in the sentence?)